#! /bin/sh -e

# DP: GPC update to build with gcc-3.4.3

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- gcc/dwarf2out.c~	2004-10-18 08:43:40.000000000 +0200
+++ gcc/dwarf2out.c	2004-10-18 09:06:45.000000000 +0200
@@ -8866,10 +8866,6 @@
 #ifdef GPC
     case REAL_CST:
     case FLOAT_EXPR:
-    case FIX_TRUNC_EXPR:
-    case FIX_CEIL_EXPR:
-    case FIX_FLOOR_EXPR:
-    case FIX_ROUND_EXPR:
     case RDIV_EXPR:
       /* In Pascal it's possible for array bounds to contain floating point
          expressions (e.g., p/test/emil11c.pas). I don't know if it's
